mautrix_instagram/portal.py

```diff
@@ -452,21 +458,34 @@ async def _reupload_instagram_media(self, source: 'u.User', media: RegularMediaI
         return await self._reupload_instagram_file(source, url, msgtype, info, intent)
 
     async def _reupload_instagram_animated(self, source: 'u.User', media: AnimatedMediaItem,
-                                           intent: IntentAPI) -> ReuploadedMediaInfo:
+                                           intent: IntentAPI) -> MediaMessageEventContent:
         url = media.images.fixed_height.webp
         info = ImageInfo(height=int(media.images.fixed_height.height),
                          width=int(media.images.fixed_height.width))
         return await self._reupload_instagram_file(source, url, MessageType.IMAGE, info, intent)
 
     async def _reupload_instagram_voice(self, source: 'u.User', media: VoiceMediaItem,
-                                        intent: IntentAPI) -> ReuploadedMediaInfo:
+                                        intent: IntentAPI) -> MediaMessageEventContent:
         url = media.media.audio.audio_src
         info = AudioInfo(duration=media.media.audio.duration)
-        return await self._reupload_instagram_file(source, url, MessageType.AUDIO, info, intent)
+        waveform = [int(p * 1000) for p in media.media.audio.waveform_data]
+        content = await self._reupload_instagram_file(source, url, MessageType.AUDIO, info, intent)
```

Comment:
So, I realized that the MSC technically requires it to be in OGG format for some reason. The files that we get from IG are mp4s, It seems to work just fine in Element, though, so I'm somewhat inclined to just leave it untranscoded.

Reply:
Instagram very likely won't be happy with ogg, so using ffmpeg for both directions would be fine: https://github.com/mautrix/facebook/commit/08097980923831d9286c9a376c486ae6a70713ef